### PR TITLE
T3C parent.config simulate topology for non topo delivery services

### PR DESCRIPTION
**Pull request by traeak**
In order to simplify the t3c parentdotconfig.go code this change creates and assigns a temporary topology on the fly for non topology delivery services.

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Control Cache Config (`t3c`, formerly ORT)

## What is the best way to verify this PR?

Generate parent.config files using this version and an older version of T3C and compare results side by side.

## PR submission checklist
- [x] This PR has tests 
- [ ] This PR has documentation - code change only
- [x] This PR has a CHANGELOG.md entry 
- [ ] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: lib/go-atscfg/parentdotconfig.go [hunk 1]
```diff
@@ -383,258 +390,112 @@ func makeParentDotConfigData(
 			continue
 		}
 
-		isMSO := ds.MultiSiteOrigin != nil && *ds.MultiSiteOrigin
+		// manufacture a topology for this DS.
```
Comment (rob05c) [inline lib/go-atscfg/parentdotconfig.go:393]: Mind putting this in a func? It's a pretty big block, and does a very singular thing. IMO it'd make the code a lot easier to read and work with.

File: lib/go-atscfg/parentdotconfig.go [hunk 2]
```diff
@@ -383,258 +390,112 @@ func makeParentDotConfigData(
...
-					pasvc.Parents = parents
-					pasvc.SecondaryParents = secondaryParents
-					pasvc.SecondaryMode = secondaryMode
-					pasvc.RetryPolicy = roundRobin
-					pasvc.GoDirect = goDirect
-					pasvc.IgnoreQueryStringInParentSelection = !*parentQStr
-					// text += `dest_domain=` + orgURI.Hostname() + ` port=` + orgURI.Port() + ` ` + parents + ` ` + secondaryParents + ` ` + roundRobin + ` ` + goDirect + ` qstring=` + parentQStr + "\n"
-				}
+		// TODO put these in separate functions. No if-statement should be this long.
+		if ds.Topology == nil || *ds.Topology == "" {
```
Comment (rob05c) [inline lib/go-atscfg/parentdotconfig.go:469]: With the fake topology you're adding above, is it ever possible for this to be true? Should this block be removed?

File: lib/go-atscfg/parentdotconfig.go [hunk 3]
```diff
@@ -165,6 +165,11 @@ func MakeParentDotConfig(
 	}, nil
 }
 
+// Check if this ds type is edge only
```
[the remark this answers is not in the document]
Comment (traeak) [reply lib/go-atscfg/parentdotconfig.go]: comment changed and moved.

File: lib/go-atscfg/parentdotconfig.go (hunk 1 above)
Comment (traeak) [reply lib/go-atscfg/parentdotconfig.go:462]: moved to CreateTopologies function although there are quite a few args and return variables :(

File: lib/go-atscfg/parentdotconfig.go (hunk 2 above)
Comment (traeak) [reply lib/go-atscfg/parentdotconfig.go]: Removed as it does seem like checks would call continue before this code would be hit.

**Review by rob05c — APPROVED** [review]
LGTM.

This is great, I've wanted to do this for ages. Big improvement in code safety and reducing duplicate logic.

**Pull request merged**